pkg/client/devops/jenkins/credential.go

```diff
@@ -237,7 +237,7 @@ func (j *Jenkins) CreateCredentialInProject(projectId string, credential *v1.Sec
 		return "", err
 	}
 
-	if response.StatusCode != http.StatusOK {
+	if response.StatusCode != http.StatusOK && response.StatusCode != http.StatusFound {
```

Comment:
what happened if the status code is `http.StatusFound`?

Reply:
I could explain why we need to check 302(http status found). 

- Firstly, some http calls to Jenkins endpoint results in 302(http status found). please see also [issue](https://github.com/jenkinsci/java-client-api/issues/7).

- Then, please see the [code change](https://github.com/kubesphere/ks-devops/pull/75/files#diff-1343812733fafb2581886180ea349d20539d7abf131b61692b8514fecf2795f8) in PR I submitted before.

Reply:
As I mentioned here https://github.com/kubesphere/ks-devops/pull/123#discussion_r685237493

All potential contributors need to know the reason why we check if the status code is `http.StatusFound`. And they might not know this PR. In another word, the codes need to explain it by themselves.

Reply:
Of course, we have responsibility to tell other contributors or code reviewers the reason why we check the status code.

Reply:
Seem like the [api doc](https://github.com/jenkinsci/credentials-plugin/blob/master/docs/user.adoc#creating-a-credentials) have not status of 302.

Reply:
Let me get it straight, any potential confusion code lines need comment.

Reply:
> Seem like the [api doc](https://github.com/jenkinsci/credentials-plugin/blob/master/docs/user.adoc#creating-a-credentials) have not status of 302. 
> 

Please see the description of [this PR](https://github.com/jenkinsci/credentials-plugin/pull/33).

Reply:
I have added the comments